mono/metadata/marshal.c

```diff
@@ -3264,6 +3265,88 @@ mono_marshal_set_callconv_from_modopt (MonoMethod *method, MonoMethodSignature *
 	}
 }
 
+static void
+mono_marshal_set_signature_callconv_from_attribute (MonoMethodSignature* sig, MonoType* cmod_type, MonoError* error)
+{
+	g_assert(cmod_type->type == MONO_TYPE_CLASS);
+	MonoClass* cmod_klass = mono_class_from_mono_type_internal (cmod_type);
+	g_assert (m_class_get_image (cmod_klass) == mono_defaults.corlib);
+	g_assert (!strcmp (m_class_get_name_space (cmod_klass), "System.Runtime.CompilerServices"));
+
+	const char* name = m_class_get_name (cmod_klass);
+	g_assert (g_str_has_prefix(name, "CallConv"));
+
+	name += strlen ("CallConv");
+	if (!strcmp (name, "Cdecl"))
+		sig->call_convention = MONO_CALL_C;
+	else if (!strcmp (name, "Stdcall"))
+		sig->call_convention = MONO_CALL_STDCALL;
+	else if (!strcmp (name, "Thiscall"))
+		sig->call_convention = MONO_CALL_THISCALL;
+	else if (!strcmp (name, "Fastcall"))
+		sig->call_convention = MONO_CALL_FASTCALL;
+	else if (!strcmp (name, "SuppressGCTransition"))
+		sig->suppress_gc_transition = 1;
+	// TODO: Support CallConvMemberFunction?
+	// TODO: Support multiple calling convetions?
```

Comment:
I played multiple calling conventions in CoreCLR and it only supports multiple calling conventions when they don't conflict.  e.g. CallConvStdCall + CallConvMemberFunction is OK, but CallConvStdCall + CallConvCdecl throws:
> System.InvalidProgramException: 'Multiple unmanaged calling conventions are specified. Only a single calling convention is supported.'